app/models/changeset/pull_request.rb

```diff
@@ -31,6 +33,17 @@ def self.find(repo, number)
     nil
   end
 
+  def self.changeset_from_webhook(project, params = {})
+    find(project.github_repo, params[:number])
+  end
+
+  def self.valid_webhook?(params)
+    data = params[:pull_request] || {}
+    return false unless data[:state] == 'open'
+
+    data[:body] =~ WEBHOOK_FILTER || data.fetch(:base, {})[:ref] == 'production'
```

Comment:
I meant in the description that it would ignore the "requirement" of having [samson] present. However, the more I think about this the more I'm thinking we should always require an explicit `[samson]` to be present. Thoughts @shajith?